src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -32,23 +32,52 @@ public unsafe static NetworkInterface[] GetLinuxNetworkInterfaces()
             const int MaxTries = 3;
             for (int attempt = 0; attempt < MaxTries; attempt++)
             {
+                // Because these callbacks are executed in a reverse-PInvoke, we do not want any exceptions
+                // to propogate out, because they will not be catchable. Instead, we track all the exceptions
+                // that are thrown in these callbacks, and aggregate them at the end.
+                List<Exception> exceptions = new List<Exception>();
```

Comment:
I moved the declaration out and lazily constructed it when an exception is encountered. Clearing isn't necessary because I'm just throwing the aggregated exception if any occurred.